tests/ui/impl-trait/in-trait/missing-static-bound-from-impl.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0310]: the associated type `<T as Original>::{opaque#0}` may not live long enough
+  --> $DIR/missing-static-bound-from-impl.rs:11:9
+   |
+LL |         Box::new(<T as Original>::f())
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |         |
+   |         the associated type `<T as Original>::{opaque#0}` must be valid for the static lifetime...
+   |         ...so that the type `impl Fn()` will meet its required lifetime bounds
+   |
+help: consider adding an explicit lifetime bound
+   |
+LL |     fn f() -> impl Fn() + 'static;
```

Comment:
I don't believe it makes sense to change the bound of the trait here. I think it makes more sense to relax the bound on the implementation. Changing the trait is like changing the source of truth -- it doesn't really seem right?

Reply:
I'm with you that the trait is the source of truth, *when the trait is non-local*. I see a lot of people define traits with incorrect bounds accidentally. Regardless, I can just remove the suggestion altogether for these cases.

Reply:
> I see a lot of people define traits with incorrect bounds accidentally.

Right, but this is a case of people falling into the "the applied fix works until i actually use it nontrivially" that we have with a lot of our suggestions. People follow them blindly, have their code compile, but then pay a bigger price later on when it turns out that the suggestion was misleading and they have to fix this bug over again.

I don't believe in general that we want to tell users to apply, e.g., `+ 'static` to their trait definition, because it's necessarily making the trait definition *less* general, and people will very likely hit this later when they're returning an RPIT that *does* carry data from something in the self type.

I think it's even more risky to do this without explaining to users that adding `+ 'static` here may necessarily break their other implementations.

Reply:
Also, in a world where we have RTN, we'd instead want to suggest this on the implementation:

```
where T::f(): 'static,
```

So we can both use that assumption in the body, and also not overly restrict trait implementations. I think for now, it's really not worth implementing this suggestion for RPITs.